tests/test-users-api.js

```diff
@@ -0,0 +1,236 @@
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+            })
+        })
+    })
+
+    describe('update user', async => {
+        it('7. should return status 200 and change user\'s information when called correctly', async() => {
+            const result = await pool.query('SELECT public."InsertUser"($1, $2, $3, $4, $5, $6, $7) as new_id', 
+            [birthdate, signupdate, name, cmnd, email, gender, expirydate])
+            const new_user_id = result.rows[0].new_id;
+            const updatedName = 'Nguyen Van C'
+            const updatedCmnd = '325323452'
+            const updatedEmail = 'c@email.com'
+            const updatedGender = true
+            const updatedBirthDate = birthdate
+            updatedBirthDate.setFullYear(birthdate.getFullYear() + 1)
+            const updatedSignUpDate = signupdate
+            updatedSignUpDate.setFullYear(signupdate.getFullYear() + 1)
+            const updatedExpiryDate = expirydate
+            updatedExpiryDate.setFullYear(expirydate.getFullYear() + 1)
+            console.log('expected date: ' + updatedExpiryDate)
+            const response = await axios.put(`${process.env.API_URL}/users/${new_user_id}`, 
+            {
+                name: updatedName,
+                cmnd: updatedCmnd,
+                email: updatedEmail,
+                gender: updatedGender,
+                birthdate: updatedBirthDate.toLocaleDateString(),
+                signupdate: updatedSignUpDate.toLocaleDateString(),
+                expirydate: updatedExpiryDate.toLocaleDateString()
+            })
+            assert.strictEqual(response.status, 200)
+            assert.strictEqual(response.data.State, State.SUCCESS)
+            assert.strictEqual(response.data.UpdatedUserId, new_user_id)
+
+            const updatedResult = await pool.query('SELECT * FROM users WHERE id = $1', [new_user_id])
+            const dbUser = updatedResult.rows[0];
+            assert.strictEqual(dbUser.name, updatedName);
+            assert.strictEqual(dbUser.cmnd, updatedCmnd);
+            assert.strictEqual(dbUser.gender, updatedGender);
+            assert.strictEqual(dbUser.birthdate.toISOString(), updatedBirthDate.toISOString());
+            assert.strictEqual(dbUser.email, updatedEmail);
+            assert.strictEqual(dbUser.signupdate.toISOString(), updatedSignUpDate.toISOString());
+            assert.strictEqual(dbUser.expirydate.toISOString(), updatedExpiryDate.toISOString());
+
+            console.log('actual date: ' + dbUser.expirydate)
+            await pool.query('CALL public."DeleteUser" ($1)', [new_user_id])
+        })
+
+        it('8. should return status 500 and State will be error when called incorrectly', async() => {
+            await axios.put(`${process.env.API_URL}/users/1`, {
+                gender: 'true'
```

Comment:
we don't need to put gender here. Do we?

Reply:
I fixed it